ui/lib/kv/addon/routes.js

```diff
@@ -6,16 +6,18 @@
 import buildRoutes from 'ember-engines/routes';
 
 export default buildRoutes(function () {
-  this.route('secrets', function () {
-    this.route('create');
-    this.route('secret', { path: '/:name' }, function () {
-      this.route('details');
+  // There are two list routes because Ember won't let a query param (e.g. *secret_prefix) be blank.
+  // We only have a query param when we're on a nested secret route like beep/boop/
```

Comment:
This is a little confusing to me. Query params seems like it'd clear up a lot of the conditional logic for routing. Could another approach be just having a default queryParam of something like `index` if we're at the root level?

Reply:
We had this very discussion and wavered on if we should pursue queryParams or the previous pattern of two routes. We ended up going two routes because it matches the URL structure folks are used to seeing now.